BDArmory/FX/ExplosionFX.cs

```diff
@@ -726,210 +792,269 @@ private void ExecutePartBlastEvent(PartBlastHitEvent eventToExecute)
             var realDistance = eventToExecute.Distance;
             var vesselMass = part.vessel.totalMass;
             if (vesselMass == 0) vesselMass = part.mass; // Sometimes if the root part is the only part of the vessel, then part.vessel.totalMass is 0, despite the part.mass not being 0.
+	        var cumulativeHPOfIntermediateParts = eventToExecute.IntermediateParts.Select(p => p.Item2).Sum();
```

Comment:
Moving the two `cumulative...OfIntermediateParts` out of the "within radius of blast" block below means that they sometimes get calculated needlessly.